### ISSUE TITLE
Mod suggestion: Tweak and Tamper

### ISSUE THREAD

**Issue by cryum** [martmists-gh/fabric-mod-wishlist#45]
### Description

Tool rework system, like Tinker (#3)  and Tetra (#18).

Emphasis is on minimal/no additional oregen, secondary features rather than stat specialization, varied combat styles, and better enchantments period. The ideas won't necessarily be original, but will hopefully integrate with each other.

Tool Composition:
Tools are composed of 4 components: Handle, Head, Binder, and Bauble. Only the first 3 are required when making a tool. If a component's max durability drops to 0, it is destroyed.

-Handle:
- Biggest contributor of Durability
- Can contain many Engravings, but few are optimal on the handle
- Passive Modifiers are usually placed here

-Head:
- Determines tool function, and base stats
- Active Modifiers are usually placed here
- Max Durability possibly affects damage/mining level
- Most expensive Engravings/Inlaying

-Binder:
- Cannot gain Modifiers
- Mainly for stats/material effects(gold for piglin flag, enchantibility, modded alloys, etc.)
- Affects blocking for weapons with defensive functions.

-Bauble:
- Allows for extra Modifiers/Engravings with minimal impact to stats
- Special Baubles can be found in treasure chests

Engraving and Inlaying:
Replaces Enchanting for applying magical effects. By inscribing a learned Rune into each component, the assembled tools can be granted various enchantments. Inlaying the runes with redstone, lapis, and other materials will increase the final enchantment level, and certain Rune combinations can apply additional effects. High enchantibility grants bonuses to these Rune combinations. Tentatively only one Rune per component, but similar runes do not stack.

Ex: Rune of Reinforcement(I, II, III, IV, V)
- Increases Max durability of component
- Good for early tools, but higher-end materials can support modifiers with their default durability.

Ex: Durandal(Reinforcement and Unbreaking)
- Rune combination from putting together components that have Reinforcement and Unbreaking
- Applies Curse of Vanishing at low Enchantibility
- Whenever Unbreaking prevents the loss of durability, regain some durability. Amount regained scales with Enchantibility

Tool Types:
Examples of various possible weapons.
- Knife: Increased chance to loot held items
- Mace/Warhammer: Rightclick to concuss the enemy, inflicting Slow. Effect scales with material/attack strength
- Greatsword: Passively adds X to Sweeping Edge level on top of enchantments. Aoe damage can exceed direct damage.
- Axe: Bonus crit damage while falling, increased chance to loot worn armor/pieces of armor
- Lunging Spear: Hop forward and attack with very long reach, but inflict Slowness on self.
- Battle Lure: Fishing rod, but can massively pull entities up or down(~10 blocks vertically)
- Shield Spear: Can be placed on the ground to act as cover and hurt enemies like a directed cactus.
- Soul Hammer: Slams a 3x3 area in front of you, and either applies a Lingering Slowness cloud, or replace the 3x3 blocks with Soul Sand
- Breaker Blade: Increased range, increased damage while falling, and a successful hit causes flying enemies to fall

Modifiers:
Modifiers are more specific tweaks, adjusting stats directly or adding secondary functions and passives to the tool. Some modifiers are inherent to different material types, but can be artificially added with the right ingredients. Each added Modifier reduces max durability by fixed values, scaling up as more are added to the same component.

Ex: Knapped Edge: Stone, Flint, Ceramics
- As Durability decreases, Mining Speed increases.
- Can be applied with diamond

Ex: Polished: Glowstone, luminecent materials
- Glows while wielded. Strength of light based on Attack Damage.
- Can be applied with sea cucumber

Ex: Honed: No innate materials
- Increase Attack Strength by 1, double durability use
- Can be applied with brick, sand and flint

Ex: Sol Brand/Luna Core
- Gains strength in the open sky during day/night time. Nether is treated as day even “under”ground, while the End is treated as night.

### Details

- Does a Forge version of this mod exist? No
- Does a Fabric version of this mod exist for an older MC version? No

**Comment by i509VCB** [comment]
Hmm this sounds a lot like the concept Workshop had:
https://github.com/CottonMC/Workshop

**Comment by cryum** [comment]
Yeah. I was helping out with that one, but the direction it's taking with overall tool composition is more complex while enchantment is simpler.
